Update release checklist: homebrew and solc-js instructions (#15494)
ethereum/solidity: ReleaseChecklist.md

 - [ ] Create a pull request in solc-bin and merge.

### Homebrew and MacOS
 - [ ] Create a local fork of the repository at ``git@github.com:Homebrew/homebrew-core.git``
 - [ ] Make sure that ``brew`` is installed on your system, or use the official Homebrew docker container:

>  ``docker run --interactive --tty --rm --pull always --volume "$(pwd):/homebrew-core" --workdir "/homebrew-core" homebrew/ubuntu22.04:latest /bin/bash``

Reviewer: ```suggestion
 ``docker run --interactive --tty --rm --pull always --volume "$PWD:/homebrew-core" --workdir /homebrew-core homebrew/ubuntu22.04``
```

>  - [ ] Make sure that ``brew`` is installed on your system, or use the official Homebrew docker container:
>  ``docker run --interactive --tty --rm --pull always --volume "$(pwd):/homebrew-core" --workdir "/homebrew-core" homebrew/ubuntu22.04:latest /bin/bash``
> #### Manual Update
>  - [ ] Open the [``solidity`` formula in Homebrew core repository](https://github.com/Homebrew/homebrew-core/blob/master/Formula/s/solidity.rb) using your preferred text editor and update the following fields:
>    - url: Update the URL to reflect the new source file for the updated version.

Reviewer: Spacing and indentation is off here:

```suggestion
 - [ ] Make sure that ``brew`` is installed on your system, or use the official Homebrew docker container:
     ``docker run --interactive --tty --rm --pull always --volume "$(pwd):/homebrew-core" --workdir "/homebrew-core" homebrew/ubuntu22.04:latest /bin/bash``

#### Manual Update
 - [ ] Open the [``solidity`` formula in Homebrew core repository](https://github.com/Homebrew/homebrew-core/blob/master/Formula/s/solidity.rb) using your preferred text editor and update the following fields:
     - url: Update the URL to reflect the new source file for the updated version.
```

>    - url: Update the URL to reflect the new source file for the updated version.
>    - sha256: Replace with the new SHA256 checksum of the source file. You can obtain it using a command like:

Reviewer: ```suggestion
   - ``url``: Update the URL to reflect the new source file for the updated version.
   - ``sha256``: Replace with the new SHA256 checksum of the source file. You can obtain it using a command like:
```

     ``curl -L https://github.com/ethereum/solidity/releases/download/v$VERSION/solidity_$VERSION.tar.gz | sha256sum``
 - [ ] Test your changes locally by running the following command from the root directory of your fork: ``brew install --build-from-source ./Formula/s/solidity.rb``
 - [ ] Open a PR in the Homebrew repository.
#### Automatic Update
 - [ ] Create a GitHub token and run the following command: ``export HOMEBREW_GITHUB_API_TOKEN=your_token_here``

>  - [ ] Run the following command to automatically update the formula and create a PR in the Homebrew repository: ``brew bump-formula-pr --version=$VERSION solidity``
>  Note: the version should not include the prefix ``v``, e.g. ``X.Y.Z``.

Reviewer: ```suggestion
 - [ ] Run the following command to automatically update the formula and create a PR in the Homebrew repository: ``brew bump-formula-pr --version=$VERSION solidity``.
     Note: the version should not include the ``v`` prefix, i.e. ``X.Y.Z``.
```

 - [ ] Create a pull request in solc-bin and merge.

### Homebrew and MacOS

>  - [ ] Update the version and the hash (``sha256sum solidity_$VERSION.tar.gz``) in the [``solidity`` formula in Homebrew core repository](https://github.com/Homebrew/homebrew-core/blob/master/Formula/s/solidity.rb).

Reviewer 2: Actually, maybe we could remove this line.

Author: Sure, it's described in the manual update step anyway.

 - [ ] Wait for the CI runs on the tag itself.
 - [ ] Take the ``solc-x.y.z.tgz`` artifact from ``build-package`` run on the tagged commit in circle-ci.
       Inspect the tarball to ensure that it contains an up-to-date compiler binary (``soljson.js``).

>        See the section "Testing version of soljson.js": https://github.com/ethereum/solidity/wiki/Building-solc%E2%80%90js-from-source#testing-version-of-soljsonjs

Reviewer: ```suggestion
       See the [Testing version of soljson.js](https://github.com/ethereum/solidity/wiki/Building-solc‐js-from-source#testing-version-of-soljsonjs) section on the wiki.
```
